test/integration/action_execution.go

```diff
@@ -314,6 +334,68 @@ func depSingleContainerWithResources(namespace, claimName string, replicas int32
 	return &dep
 }
 
+func createBarePod(client *kubeclientset.Clientset, pod *corev1.Pod) (*corev1.Pod, error) {
+	var errInternal error
+	var barepod *corev1.Pod
+	if err := wait.PollImmediate(framework.PollInterval, framework.TestContext.SingleCallTimeout, func() (bool, error) {
+		barepod, errInternal = client.CoreV1().Pods(pod.Namespace).Create(context.TODO(), pod, metav1.CreateOptions{})
+		if errInternal != nil {
+			glog.Errorf("Unexpected error while creating pod: %v", errInternal)
+			return false, errInternal
+		}
+		return true, nil
+	}); err != nil {
+		return nil, err
+	}
+
+	return waitForBarePod(client, barepod.Name, barepod.Namespace)
+}
+
+func depBarePodWithResources(namespace, claimName string, replicas int32, withVolume bool) *corev1.Pod {
+	barePod := corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "test-",
+			Namespace:    namespace,
+		},
+
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:    "test-cont",
+					Image:   "busybox",
+					Command: []string{"/bin/sh"},
+					Args:    []string{"-c", "while true; do sleep 30; done;"},
+					Resources: corev1.ResourceRequirements{
+						Limits: map[corev1.ResourceName]resource.Quantity{
+							corev1.ResourceCPU:    resource.MustParse("100m"),
+							corev1.ResourceMemory: resource.MustParse("200Mi"),
+						},
+						Requests: map[corev1.ResourceName]resource.Quantity{
+							corev1.ResourceCPU:    resource.MustParse("50m"),
+							corev1.ResourceMemory: resource.MustParse("100Mi"),
+						},
+					},
+				},
+			},
+		},
+	}
+
+	if withVolume {
```

Comment:
Why not add a test for volume based bare pod moves also, given you have already added the code to have a volume with the pod.